nexus/src/db/error.rs

```diff
@@ -183,10 +183,20 @@ where
         DieselError::DatabaseError(kind, info) => {
             PublicError::internal_error(&format_database_error(kind, &*info))
         }
-        error => PublicError::internal_error(&format!(
-            "Unknown diesel error: {:?}",
-            error
-        )),
+        error => {
+            let context = match make_not_found_error() {
+                PublicError::ObjectNotFound { type_name, lookup_type } => {
+                    format!("looking up {:?} {:?}", type_name, lookup_type)
```

Comment:
Maybe a nit, but is it true that this error is _only_ hit when "looking up" a resource? I think your suggestion of a trait that provides an additional context method is really attractive, since it would allow us to provide this string or some other information from the caller.

Reply:
I guess it depends on what you mean by "lookup".  I believe it's always used in a context where we expect a specific object to exist and it doesn't.  Sometimes that's because we're trying to update it or delete it, not fetch it.  I could make the language more general ("accessing object") to avoid confusion.